src/build/dotnet/steps/net-core.steps.yaml

```diff
@@ -36,6 +36,13 @@ steps:
       command: restore
       projects: ${{parameters.projects}}
       vstsFeed: 'Audacia.Public/AudaciaPublic'
+      
```

Comment:
I'm not sure adding a second restore step here is the right move. There needs to be a decision as to whether these steps should exactly replicate the old repository (in which case it should be left as-is) or whether we make enhancements at this point.

If enhancements are the table then I am aware that the restore step can be problematic, but I wonder if it would be better to either restore all .csproj files, or add an extra parameter to provide a pattern for projects to restore.

Any thoughts @jackpercy-acl / @eddie-carr?

Reply:
We might be able to do the restore like so:

```yaml
projects: |
  ${{parameters.projects}}
  ${{parameters.tests}}
```

Might need to handle the case when `shouldRunTests` is set to false, perhaps like

```yaml
${{ if eq(parameters.shouldRunTests, true) }}:
  projects: |
    ${{parameters.projects}}
    ${{parameters.tests}}
${{ else }}:
  projects: ${{parameters.projects}}
```

Reply:
I did a quick test of the above and looks like this will work

Reply:
good solution @jackpercy-acl

Reply:
Thanks, I'd considered the issue of the test project being blank didn't know the syntax for doing it on the Test project optionally based on if testing, that is probably sufficient without needing a empty string check on the test proj variable as the test step would fail then anyway. This syntax does work for combining the two restores.

The need to restore the test project here is needed now because of the Audacia Seed reference, and that it doesn't look in the Audacia source (the restore step having vstsFeed resolves that issue it seems, but doesn't work doing restore as part of the build step). I can't see any other way to resolve this as our project seems set up ok, however it is odd that only this package has caused the error and not Audacia.CodeAnalysis which had been working in the test project previously without this issue occurring - so if anyone has any alternatives to make this step not required please do suggest!

error NU1101: Unable to find package Audacia.Seed.EntityFrameworkCore. No packages exist with this id in source(s): C:\Program Files\dotnet\library-packs, Microsoft Visual Studio Offline Packages, nuget.org
  Failed to restore D:\a\1\s\tests\Persimmon.DocumentManagement.Tests\Persimmon.DocumentManagement.Tests.csproj (in 20.93 sec).